src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/Actions/EliminationForms/FailedToQualifyForm.razor

```diff
@@ -4,18 +4,19 @@
 @using NTS.Judge.Blazor.Ports
 @using Not.Structures
 
-@inherits NotComponent
+@inherits EliminationForm
 
-<MudSelect T="string" Label="Failed to Qualify Reason" MultiSelection="true" Delimiter="+" SelectedValues="@FTQReason" SelectedValuesChanged="SetInputState" Placeholder="Select a reason">
-    @foreach (var failToQualifyCode in SelectListModel.SelectListFromEnum<FTQCodes>())
+<MudSelect T="FTQCodes" Label="Failed to Qualify Reason" MultiSelection="true" Delimiter="+"
+    SelectedValues="@Codes" SelectedValuesChanged="SetInputState" Placeholder="Select a reason">
```

Comment:
`T="FTQCodes"` isn't  necessary as it can be auto-inferred by the usage of `SelectedValues`.

Furthermore you can use `@bind-SelectedValues` and remove `SelectedValuesChanged` here as well.

Reply:
What about this one?